sdlib/d/gc/extent.d

```diff
@@ -266,7 +284,64 @@ public:
 		// FIXME: in contract.
 		assert(isSlab(), "slabData accessed on non slab!");
 
-		return _metadata.slabData;
+		return _metadata.slabData.slabData;
+	}
+
+	/**
+	 * Freespace Flag features for slabs.
+	 */
+
+	@property
+	bool supportsFreeSpace() const {
+		return isAppendableSizeClass(sizeClass);
+	}
+
+	bool mayHaveFreeSpace(uint index) {
+		assert(isSlab(), "mayHaveFreeSpace accessed on non slab!");
+
+		return supportsFreeSpace && slabData.valueAt(index);
+	}
```

Comment:
I don't think this is particularly useful, trying to set free space on a non allocated slab is clearly a bug, not something you want to check and bail out of.